lib/facility.rb

```diff
@@ -1,14 +1,81 @@
+require "date"
+
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, 
+                :address, 
+                :phone, 
+                :services,
+                :hours, 
+                :registered_vehicles, 
+                :collected_fees,
+                :holidays_closed
```

Comment:
Similarly here.. it doesn't look like you are trying to update these attributes outside of this class, so I thin kthey can all be set as readers.